src/components/Toast/Toast.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from "react";
+import Snackbar from "@material-ui/core/Snackbar";
+import IconButton from "@material-ui/core/IconButton";
+import CloseIcon from "@material-ui/icons/Close";
+import Box from "@material-ui/core/Box";
+
+export enum ToastStatus {
+  Success = "#81c784",
+  Error = "#e57373",
+  Warning = "#ffb74d",
+  Info = "#64b5f6"
+}
+
+interface CreateToast {
+  message: React.ReactNode;
+  status?: ToastStatus
+  onClose?() :void;
+}
+
+interface ToastContext {
+  create(props: CreateToast): void;  
+}
+
+const ToastContext = React.createContext<ToastContext>({ create: () => {} });
+export function useToastContext() {
+  return React.useContext(ToastContext);
+}
+
+export const ToastContextProvider: React.FC = ({ children }) => {
+  const [toastProps, setToastProps] = React.useState<CreateToast>();
+  const create = React.useCallback((props: CreateToast) => {
+    setToastProps(props);
+  }, [setToastProps]);
+
+  const onClose = React.useCallback(() => {
+    setToastProps(undefined);
+    toastProps?.onClose();
+  }, [toastProps?.onClose, setToastProps])
+
+  return (
+    <ToastContext.Provider value={{ create }}>
+      <Snackbar 
+        open={!!toastProps} 
```

Comment:
This should also check for message, the required field. Then the or null can be fixed on line 52